app/src/main/java/ch/epfl/sweng/rps/ui/statistics/StatisticsFragment.kt

```diff
@@ -123,8 +152,88 @@ class StatisticsFragment : Fragment() {
         statsTable.addView(row)
 
     }
+
+
+    private fun readStatsData(userID: String, firebaseCallBack: FirebaseCallBack) {
```

Comment:
You should not handle it inside a fragment, wrap it in another class and also make sure you are not reinventing the wheel! We have a firebase handler use it and modify it according to your needs if needed.

Reply:
Sure, I realized this problem and communicated with Gaetan as well. I will update after meeting with you guys because I don't know where should I start to deal with it since I have no much experience in FireBase and software architecture design :(